### PR TITLE
Tests for store_cli.go

### PR DESCRIPTION
**Pull request by pranavrc**

### CODE REVIEW

File: store-cli_test.go [hunk 1]
```diff
@@ -1,744 +1,55 @@
...
+	expected = "http://store.screwdriver.cd/v1/builds/10038-test"
+	if i.String() != expected {
+		t.Fatalf("Expected '%s' but '%s'", expected, i)
+	}
+
+	storeType = "artifacts"
+	scope = "events"
+	key = "test"
+	i, _ = makeURL(storeType, scope, key)
+	expected = "http://store.screwdriver.cd/v1/builds/events/499/test"
```
Comment (d2lam) [inline store-cli_test.go]: Interesting. think we need to add some extra validation. 
If folder is `builds`, then it shouldn't append the `scope`. 
This combination shouldn't be allowed. This case should be `builds/{buildID}-ARTIFACTS`

File: store-cli_test.go [hunk 2]
```diff
@@ -1,744 +1,55 @@
...
+	expected := "http://store.screwdriver.cd/v1/caches/events/499/cache-1"
+	if i.String() != expected {
+		t.Fatalf("Expected '%s' but '%s'", expected, i)
+	}
+
+	storeType = "artifacts"
+	scope = "builds"
+	key = "test"
+	i, _ = makeURL(storeType, scope, key)
+	expected = "http://store.screwdriver.cd/v1/builds/10038-test"
```
Comment (d2lam) [inline store-cli_test.go]: This case should be `builds/{buildID}-ARTIFACTS/test`

File: store-cli_test.go [hunk 3]
```diff
@@ -1,744 +1,55 @@
...
+
+	storeType := "cache"
+	scope := "events"
+	key := "cache-1"
+	i, _ := makeURL(storeType, scope, key)
+	expected := "http://store.screwdriver.cd/v1/caches/events/499/cache-1"
+	if i.String() != expected {
+		t.Fatalf("Expected '%s' but '%s'", expected, i)
+	}
+
```
Comment (d2lam) [inline store-cli_test.go:29]: Can we add one for `logs`?

File: store-cli.go [hunk 4]
```diff
@@ -43,62 +42,70 @@ func makeURL(storeType, scope, key string) (*url.URL, error) {
...
-	switch scope {
-	case "events":
-		path = "events/" + os.Getenv("SD_EVENT_ID") + "/" + key
-	case "builds":
-		path = os.Getenv("SD_BUILD_ID") + "-" + key
+	var fullpath string
+	if len(path) > 0 {
+		fullpath = fmt.Sprintf("%s/%s/%s", storeURL, version, path)
+	} else {
+		fullpath = fmt.Sprintf("%s/%s", storeURL, version)
```
Comment (d2lam) [inline store-cli.go]: when would we have this case?

Comment (pranavrc) [reply store-cli.go]: When it's an invalid call? We don't want arbitrary URLs being built based on whatever is being passed in, I suppose

Comment (d2lam) [reply store-cli.go]: Would returning error be better?

Comment (pranavrc) [reply store-cli.go]: Yeah working on the error/validation part. Thought I'd make it a separate PR but I'll push it to this PR as well

**Review by d2lam — APPROVED** [review]

**Pull request merged**